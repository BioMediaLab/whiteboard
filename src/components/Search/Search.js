import React from 'react'
import TextField from '../TextField'
import Button from '../Button'

import './Search.css'

export default function Search({ id, label, name, onChange, onSubmit, value }) {
  const handleSubmit = e => {
    e.preventDefault()
    onSubmit()
  }
  return (
    <form className="Search" onSubmit={handleSubmit}>
      <TextField
        id={id}
        label={label}
        name={name}
        onChange={onChange}
        type="search"
        value={value}
      />
      <Button type="submit">search</Button>
    </form>
  )
}
