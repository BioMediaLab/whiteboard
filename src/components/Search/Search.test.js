import React from 'react'
import Search from './'

describe('Search', () => {
  test('can render', () => {
    const component = render(
      <Search
        id="example"
        name="example"
        label="example"
        value="a test value"
        onChange={() => {}}
      />
    )

    expect(component).toMatchSnapshot()
  })

  test('can handle submit', () => {
    const onSubmit = jest.fn()
    const component = render(
      <Search
        id="example"
        name="example"
        label="example"
        value="just a test"
        onChange={() => {}}
        onSubmit={onSubmit}
      />
    )

    component.clickText('search')

    expect(onSubmit).toHaveBeenCalled()
  })
})
