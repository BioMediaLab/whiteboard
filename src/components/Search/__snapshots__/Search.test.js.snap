// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search can render 1`] = `
<div>
  <form
    class="Search"
  >
    <div
      class="TextField"
    >
      <label
        for="example"
      >
        example
      </label>
      <input
        id="example"
        name="example"
        type="search"
        value="a test value"
      />
    </div>
    <button
      class="Button"
      type="submit"
    >
      search
    </button>
  </form>
</div>
`;
